editor/debugger/editor_visual_profiler.cpp

```diff
@@ -747,6 +752,13 @@ EditorVisualProfiler::EditorVisualProfiler() {
 	clear_button->connect(SceneStringName(pressed), callable_mp(this, &EditorVisualProfiler::_clear_pressed));
 	hb->add_child(clear_button);
 
+	autostart_checkbox = memnew(CheckBox);
+	autostart_checkbox->set_toggle_mode(true);
+	autostart_checkbox->set_text(TTR("Autostart"));
+	autostart_checkbox->set_pressed(EditorSettings::get_singleton()->get_project_metadata("debug_options", "autostart_visual_profiler", false));
+	autostart_checkbox->connect(SceneStringName(pressed), callable_mp(this, &EditorVisualProfiler::_autostart_pressed));
```

Comment:
You could connect to `toggled` signal, which would also allow to make `autostart_checbkox` local.